pkg/entrypoint/server_test.go

```diff
@@ -0,0 +1,197 @@
+package entrypoint
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+
+	"github.com/go-logr/logr"
+	"github.com/go-logr/zapr"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"go.uber.org/zap"
+)
+
+const apiAddr = "localhost:8080"
+
+type testCase struct {
+	input    []Job
+	expected []statusResponse
+}
+type statusResponse struct {
+	Jobs []job
+}
+type job struct {
+	Name   string
+	Status string
+}
+
+var log logr.Logger
+var _ = BeforeSuite(func() {
+	zapLog, _ := zap.NewDevelopment()
+	log = zapr.NewLogger(zapLog)
+})
+
+var _ = Describe("entrypoint status API test", func() {
+	It("should state of successful command changed from running to completed", func() {
```

Comment:
Those `It`s have same code and different test case data.

How about writing with loop like https://github.com/cybozu-go/meows/blob/v0.7.0/controllers/runner_manager_test.go#L41 ?

Reply:
note: Those `It`s use `defer` inside, so we have to wrap them with `func(){ ... }()`.